config/routes.rb

```diff
@@ -9,9 +9,10 @@
 
   namespace :v0, defaults: { format: 'json' } do
     resources :in_progress_forms, only: [:index, :show, :update, :destroy]
-    resources :letters, only: [:index, :show] do
+    resources :letters, only: [:index] do
       collection do
         get 'beneficiary', to: 'letters#beneficiary'
+        post ':id', to: 'letters#download'
```

Comment:
I don't understand what the number of endpoints has to do with using `POST` - couldn't it be a `GET` with url query params?

Also, what endpoints you are speaking of?